symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php

```diff
@@ -23,43 +23,49 @@
  * Description of GoogleAuthProvider
  */
 class GoogleAuthProvider extends AbstractAuthProvider {
-
     const PROFILE_SCOPE = 'profile';
     const EMAIL_SCOPE = 'email';
+    protected $loginService = null;
+    protected $option = array();
 
-    public function validateUser($provider, $authProvider = null) {
-
+    /**
+     * @param OpenidProvider $provider
+     * @param AuthProviderExtraDetails $authProvider
+     * @return array
+     */
+    public function validateUser($provider, $authProvider = null)
+    {
         $gClient = new Google_Client();
         $gClient->setApplicationName('Login to Google +');
         $gClient->setClientId($authProvider->getClientId());
         $gClient->setClientSecret($authProvider->getClientSecret());
         $gClient->setRedirectUri($provider->getProviderUrl());
         $gClient->setDeveloperKey($authProvider->getDeveloperKey());
-        $gClient->addScope(array(self::EMAIL_SCOPE,self::PROFILE_SCOPE));
-
-        $gService = new Google_Service_Plus($gClient);
+        $gClient->addScope(array(self::EMAIL_SCOPE, self::PROFILE_SCOPE));
+        $requestParameters = $this->getOption();
+        $requestCode = $requestParameters['code'];
 
-        if (isset($_GET['code'])) {
-            $gClient->authenticate($_GET['code']);
+        if (isset($requestCode)) {
```

Comment:
requestCode is already initiated .. so why checking again ?

Reply:
Actually, fetchAccessTokenWithAuthCode() function call twice at a time.first time requestCode is empty then isset() return true and call fetchAccessTokenWithAuthCode() function and redirect to same page with URL value. after that requestCode set and ignore the if condition we can use isset() and is_empty()both but google recommend isset();